frontend/src/components/git-dialog.tsx

```diff
@@ -0,0 +1,241 @@
+import { client } from "@/client";
+import { cn } from "@/lib/utils";
+import { useStore, useUiStore } from "@/store";
+import type { Commit, Diff } from "@/types/api-types";
+import type { AppState, UiState } from "@/types/state";
+import { formatDistanceToNow } from "date-fns";
+import { GitBranch, GitCompareArrows } from "lucide-react";
+import { useTheme } from "next-themes";
+import React from "react";
+import ReactDiffViewer from "react-diff-viewer-continued";
+import Markdown from "react-markdown";
+import { toast } from "sonner";
+import { useShallow } from "zustand/react/shallow";
+import { CopyButton } from "./action-button";
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogHeader,
+  DialogTitle,
+} from "./ui/dialog";
+import { ScrollArea } from "./ui/scroll-area";
+import { Tabs, TabsContent, TabsList, TabsTrigger } from "./ui/tabs";
+
+interface CommitDetailsProps {
+  commit: Commit | null;
+}
+
+const CommitDetails = ({ commit }: CommitDetailsProps) => {
+  if (!commit) {
+    return (
+      <div className="text-muted-foreground text-sm text-center">
+        Select a commit to see its details
+      </div>
+    );
+  }
+  return (
```

Comment:
Stretch goal: 
We could also display the tags of a commit as nice little badges